packages/gatsby/src/utils/api-node-helpers-docs.js

```diff
@@ -0,0 +1,293 @@
+/* eslint-disable no-unused-vars */
+
+/** */
+const GatsbyReporter = {
+  /**
+   * @callback GatsbyReporterFn
+   * @param {string} message Message to display
+   * @returns {void}
+   */
+
+  /**
+   * @callback GatsbyReporterFnWithError
+   * @param {string} message Message to display
+   * @param {Error}[error] Optional error object
+   * @returns {void}
+   */
+
+  /**
+   * @type {GatsbyReporterFn}
+   * @example
+   * reporter.info(`text`)
+   */
+  info: true,
+
+  /**
+   * @type {GatsbyReporterFn}
+   * @example
+   * reporter.warn(`text`)
+   */
+  warn: true,
+
+  /**
+   * @type {GatsbyReporterFnWithError}
+   * @example
+   * reporter.error(`text`, new Error('something'))
+   */
+  error: true,
+
+  /**
+   * @type {GatsbyReporterFnWithError}
+   * @example
+   * reporter.panic(`text`, new Error('something'))
+   */
+  panic: true,
+
+  /**
+   * @type {GatsbyReporterFnWithError}
+   * @example
+   * reporter.panicOnBuild(`text`, new Error('something'))
+   */
+  panicOnBuild: true,
+};
+
+/** */
+const GatsbyCache = {
+  /**
+   * @param {string} key
+   * @returns {Promise<any>}
+   * @example
+   * const value = await cache.get(`unique-key`)
+   */
+  get: true,
+
+  /**
+   * @param {string} key
+   * @param {any} value
+   * @returns {Promise<any>}
+   * @example
+   * await cache.set(`unique-key`, value)
+   */
+  set: true,
+};
+
+/** */
+const GatsbyTracing = {
+  /**
+   * Global tracer instance. Check
+   * [opentracing Tracer documentation](https://opentracing-javascript.surge.sh/classes/tracer.html)
+   * for more details.
+   * @type {Opentracing.Tracer}
+   */
+  tracer: true,
+
+  /**
+   * Tracer span representing API run. Check
+   * [opentracing Span documentation](https://opentracing-javascript.surge.sh/classes/span.html)
+   * for more details.
+   * @type {Opentracing.Span}
+   */
+  parentSpan: true,
+
+  /**
+   * @callback GatsbyTracing.StartSpan
+   * @param {string} spanName name of the span
+   * @returns {Opentracing.Span}
+   */
+
+  /**
+   * Convience function to start span using API run span as parent.
```

Comment:
```suggestion
   * Convenience function to start a span using API run span as parent.
```

Fixed a typo, but I don't understand the sentence. Maybe we could re-phrase it?

Reply:
Thanks for the example, that makes sense. Hmmmm this is a tricky one - how about:

Start a tracing span. The span will be created as a child of the currently running API span.
This is a convenience wrapper for `` tracing.tracer.startSpan(`span name`, { childOf: tracing.parentSpan}) ``.